docs/howto/write-unit-tests-for-a-charm.md

```diff
@@ -162,6 +162,50 @@ def test_charm_runs(my_charm):
 If you use pytest, you should put the `my_charm` fixture in a top level `conftest.py`, as it will likely be shared between all your unit tests.
 ```
 
+## Reusing state
+
+Each test is typically an isolated test of how your charm responds to a single event. If you need to use the same input state for several events, we recommend using a fixture that produces the input state. For example:
+
+```python
+@pytest.fixture
+def state_normal():
+    container = testing.Container("some-container", can_connect=True)
+    return testing.State(
+        containers={container},
+        config={"server-port": 8080},
+        leader=True,
+    )
+
+
+def test_config_changed(state_normal):
+    ctx = testing.Context(Charm)
+    state_out = ctx.run(ctx.on.config_changed(), state_normal)
+    # Check the output state.
+
+
+def test_status(state_normal):
```

Comment:
Update status is a very special event. If we use it, we have to be clear that this is used as a kludge for Delta charms not being able to process the actual juju event previously at that point in time.

What bothers me is that we are suggesting our users to cover all of events that they subscribe to. Which would be a form of useless coverage in unit testing. What we should be doing instead is to steer them towards testing interesting state transitions.